packages/browser-wallet/src/popup/shared/IdCard/IdCard.tsx

```diff
@@ -100,12 +121,21 @@ export default function IdCard({ name, provider, status, onNameChange, className
             }}
         >
             <header className="id-card__header">
-                <ConcordiumIcon />
+                <IdentityStatusIcon status={status} />
                 {t('id.header')}
+                {onNameChange && (
+                    <Button
+                        disabled={isEditing}
+                        clear
+                        className="id-card__edit-button"
+                        onClick={() => setIsEditing(true)}
+                    >
+                        <EditIcon />
+                    </Button>
+                )}
             </header>
             <div className="id-card__name">
-                {!onNameChange && <div className="id-card__name-form">{name}</div>}
-                {onNameChange && <EditableName name={name} onChange={onNameChange} />}
+                <EditableName name={name} onChange={handleNameChange} isEditing={isEditing} />
```

Comment:
The checkmark feels a bit weird after the text instead of replacing the edit button :shrug:

Reply:
Yeah I agere.. I'll go over it with @jens-concordium

Reply:
Moved the checkmark to the location of the edit button.